docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -0,0 +1,698 @@
+
+=== RESTEasy Integration
+
+If you want to use Qute in your JAX-RS application, you'll need to add the `quarkus-resteasy-qute` extension first.
+In your `pom.xml` file, add:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-resteasy-qute</artifactId>
+</dependency>
+----
+
+This extension registers a special `ContainerResponseFilter` implementation so that a resource method can return a `TemplateInstance` and the filter takes care of all necessary steps.
+A simple JAX-RS resource may look like this:
+
+.HelloResource.java
+[source,java]
+----
+package org.acme.quarkus.sample;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+import io.quarkus.qute.TemplateInstance;
+import io.quarkus.qute.Template;
+
+@Path("hello")
+public class HelloResource {
+
+    @Inject
+    Template hello; <1>
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public TemplateInstance get(@QueryParam("name") String name) {
+        return hello.data("name", name); <2> <3>
+    }
+}
+----
+<1> If there is no `@ResourcePath` qualifier provided, the field name is used to locate the template. In this particular case, we're injecting a template with path `templates/hello.txt`.
+<2> `Template.data()` returns a new template instance that can be customized before the actual rendering is triggered. In this case, we put the name value under the key `name`. The data map is accessible during rendering. 
+<3> Note that we don't trigger the rendering - this is done automatically by a special `ContainerResponseFilter` implementation.
+
+==== Variant Templates
+
+Sometimes it could be useful to render a specific variant of the template based on the content negotiation.
+`VariantTemplate` is a perfect match for this use case:
+
+[source,java]
+----
+@Path("/detail")
+class DetailResource {
+  
+  @Inject
+  VariantTemplate item; <1>
+
+  @GET
+  @Produces({ MediaType.TEXT_HTML, MediaType.TEXT_PLAIN })
+  public Rendering item() {
+     return item.data(new Item("Alpha", 1000)); <2>
+  }
+}
+----
+<1> Inject a variant template with base path derived from the injected field - `src/main/resources/templates/item`.
+<2> The resulting output depends on the `Accept` header received from the client. For `text/plain` the `src/main/resources/templates/item.txt` template is used. For `text/html` the `META-INF/resources/templates/item.html` template is used.
+
+
+=== Development Mode
+
+In the development mode, all files located in `src/main/resources/templates` are watched for changes and modifications are immediately visible.
+
+=== Configuration
+
+* `quarkus.qute.suffixes` - a set of suffixes used when attempting to locate a template; by default, `engine.getTemplate(foo)` would result in several lookups: `src/main/resources/templates/foo`, `src/main/resources/templates/foo.html` and `src/main/resources/templates/foo.txt`
```

Comment:
I would use the configuration reference include and add more information to the javadoc if you need more.